### ISSUE TITLE
Release .48 new function "Download Reblogged Posts" not working right

### ISSUE THREAD

**Issue by tringate** [johanneszab/TumblThree#81]
I discovered that if you download a blog with "download reblogged posts" turned on, then scan the blog deleting duplicated, and then rescan the blog the same reposts are again downloaded.

Unlike other downloads which have been previously downloaded tumblr does not attempt to download the same post again.

I have tried this several times.  If I turn this function off and do a re-scan it does not download those files.
If I turn it back on and do another re-scan it once again downloads those reblogged posts.

It appears this function over rides the test if a post has already been downloaded by this blog already, or it is not registering that a re-blog post has already been processed and should be marked as downloaded.

**Comment by johanneszab** [comment]
I don't get it. What do you mean with "deleting duplicates"? The reblogged posts? The statistics in the _Details_ view do of course change if you turn on/off the _download reblogged posts_ option. It only counts the the posts for the current applied settings. Previously I've counted all the posts the grabber found, regardless of any settings, and people complained that TumblThree does not download everything ..

For me it works here. I've just tested downloading a newly added blog (http://wallpaperfx.tumblr.com and http://mywallpapercollections.tumblr.com), turning on the _download reblogged posts_ option and downloading everything. If I delete all the files afterwards, put the blog back into the queue, nothing gets downloaded. 
The same is true if I turn the _download reblogged posts_ off, and download it again. If I then turn the option on, only the rebloggs are downloaded.

**Comment by tringate** [comment]
Let me explain,

I do a download of a new blog, I turn on this function and everything is downloaded as expected.

Then I use an offline duplicate image finder that compares the actual photo to other photos and when duplicates are found they are deleted essentially removing duplicate files from the blog.

Normally a rebolg is of a photo in some other blog so it would not be duplicated in this blog, but often people reblog the same photo into a blog that is from a different blog which had that photo also so now two duplicates of the same photo plus the original which might have been a reblog itself are now in this "new" blog.

It is not uncommon at all for me to find as many as 20-30 of the same photo in a set of only 10 blogs.

In your test I believe you said that when you turned  the re-blog function back on, it once again downloaded the blogs it had already downloaded.  That is exactrly what I am seeing.

If I leave that function enabled, download the blog, then offline delete those reblog photos, then launch tumble3 and again scan the same blog those same rebloged photos are once again downloaded.

I know it is confusing, but presently I have over 3 million files of photos and the duplicates are a huge number in that many.  By deleting them I remove several gigabytes of photos that are nothing more than duplicates of the same photo.

I will try the two blogs you tested to see if I can download those using my process and if they do or do not cause the same issue.

I will post my results here.

Thanks

**Comment by johanneszab** [comment]
> If I leave that function enabled, download the blog, then offline delete those reblog photos, then launch tumble3 and again scan the same blog those same rebloged photos are once again downloaded.

No, thats not happening for me.

Do you have any small blog where that happens? Send me your .tumblr and _files.tumblr files for that particular blog.

**Comment by tringate** [comment]
I agree, I just tested your example and it does not fail for me either.'
I found and deleted these 6 duplicates, and never again did they try to be downloaded.

"D:\Tumblr\Blogs\wallpaperfx\tumblr_n0qre8QEZA1rgdnd6o1_1280.jpg","5/21/2017 12:31:58 PM
"D:\Tumblr\Blogs\wallpaperfx\tumblr_n0qrmhOhjB1rgdnd6o1_1280.jpg","5/21/2017 12:31:59 PM

"D:\Tumblr\Blogs\wallpaperfx\tumblr_mmj7uoDLWh1rgdnd6o1_1280.jpg","5/21/2017 12:32:07 PM
"D:\Tumblr\Blogs\wallpaperfx\tumblr_mlr1u1onvt1rgdnd6o1_1280.jpg","5/21/2017 12:32:09 PM

"D:\Tumblr\Blogs\wallpaperfx\tumblr_mmj7uoDLWh1rgdnd6o2_1280.jpg","5/21/2017 12:32:08 PM
"D:\Tumblr\Blogs\wallpaperfx\tumblr_mlurarBPiq1rgdnd6o1_1280.jpg","5/21/2017 12:32:08 PM

"D:\Tumblr\Blogs\wallpaperfx\tumblr_mmj7uoDLWh1rgdnd6o4_1280.jpg","5/21/2017 12:32:08 PM
"D:\Tumblr\Blogs\wallpaperfx\tumblr_mlurarBPiq1rgdnd6o2_1280.jpg","5/21/2017 12:32:08 PM

"D:\Tumblr\Blogs\wallpaperfx\tumblr_mmj7uoDLWh1rgdnd6o3_1280.jpg","5/21/2017 12:32:07 PM
"D:\Tumblr\Blogs\wallpaperfx\tumblr_mlurarBPiq1rgdnd6o4_1280.jpg","5/21/2017 12:32:09 PM

"D:\Tumblr\Blogs\mywallpapercollections\tumblr_mhr2q2f9lm1rkcocmo1_400.jpg","5/21/2017 12:31:48 PM
"D:\Tumblr\Blogs\mywallpapercollections\tumblr_mfy2a6gH7x1rkcocmo1_400.jpg","5/21/2017 12:31:49 PM

Let me try a total new download of the blog I first saw this happen to.

I will delete the entire blog and then do a full download of it with the download reblog turned on.  I will delete the duplicates from that blog, and then do another scan leaving the reblog turned on.

That is what I had done with an existing blog and the reblogs continued to be downloaded again each time I did the delete.

It could be some problem because it was an existing blog.  I also tried turning on the "force rescan" to that same failing blog expecting it to download the entire blog again which has some 1,100 files in it and my downloaded directory had only 200 files in it.  After the rescan, there were still only 200 files in the directory, not 1,100 as I expected.

Maybe I don't understand what that function does, so do not link that to the re-blogs issue I am having.  I only mentioned it because it might also be stumbling over something that is wrong with my existing index files.

I will look for a small blog that fails, if I am able to recreate the failure using an entirely new copy of the blog I say this failure with.

**Comment by johanneszab** [comment]
Maybe in your _files.tumblr in the Links array [ ] where the filenames are stored is something else inside, except for filenames or numbers. If so, the check for already downloaded files maybe fails. I'm not sure if it would.

You could briefly scan one of your files. If you always used the latest binary and updated your .tumblr files from each previously release, I wouldn't guarantee that I've not uploaded once a broken release. You can compare it to a fresh one to see what's usually stored there.

**Comment by tringate** [comment]
What I have discovered is that when I did the “force rescan” of the blogs it made this problem go away.  I can only assure that something was wrong
with the index files because now after several rescans of many blogs I have not seen a single repeat of this issue.

I tried to recreate the original problem with that blog and could not do so.

While checking the index files I discovered you have the locations paths defined in them.  I have the urge to attempt to add the “group” function by simply editing those
values, and moving the photos to the appropriate folder path.

I suspect that would not work, but it does suggest maybe an easy solution.

Is it possible to make the root folder for the blogs be defined for each blog in the blog section?

My idea for the group kind of addition was to add a new column to the display maybe next to “tags” and call it “group” which would be the group that blog was a part of.
then when tumblr3 built the file paths put that value as an addition to the path such as: \blog\group\blog-name.

What I do now is create a unique instance of tumblr3 for each group and run them one at a time to keep groups of files all together.
